refactor: add unified modal layer

ThatNerdSquared opened the pull request:
### What Issue does this PR resolve? (Link to GitHub Issue, approved features and bugs will be given priority)
This PR implements the following...

bugfixes/enhancements:
- prevent buttons from scrolling off screen in section info modal, if section metadata is very long (BIOC_V 303 is a good course to test this)
- destructive actions in modals (e.g. remove section) are now coloured as red
- improved consistency in modal design across the extension

code cleanups:
- replace the following similar modal components with a unified, configurable `ModalWindow` component:
    - `ConfirmationModal` (shown when "clear worklist" button is pressed)
    - `SectionPopup` (shown when sections are selected)
    - `InfoModal` (shown when users click question mark icons in settings)
- added a `ModalLayer` wrapper component that manages showing all modals from a single top-level layer
    - components can now post a modal by adding a config in [`modalReducer`](https://github.com/ThatNerdSquared/workday-calendar-extension/blob/d0e40a4849ee5322c9f4a1bdc2f0610896e2d364/src/content/ModalLayer.tsx#L59) and then calling `dispatchModal` using the provided context
    - this is hard to explain, but see the `Form.tsx` diff to see how it looks at the callsite
- remove redundant code in `Settings.tsx` that is already present in `Tools.tsx`
- add some CSS variables in `index.css` to reuse common colours, sizes, etc
- ran prettier on the majority of files I touched (sorry about that, it seems to have inflated the diffs)


### Please provide a video demo below, or a screenshot and description of the change.

https://github.com/mlool/workday-calendar-extension/assets/72814106/a8b9b36e-f949-4b33-b80e-6e21e11a3c6c


### Tag reviewers for the PR below.
@mlool 
NOTE: `SectionInfoBody.tsx` is really just `SectionPopup.tsx` with some changes, and renamed - unfortunately there are enough differences that git thinks they're completely different files. Sorry about that.

jaskirat-gill commented:
Couple nitpicks while we're refactoring
1. Can we add the same padding to the discord button as the rest of the settings?
![image](https://github.com/mlool/workday-calendar-extension/assets/60302907/dbda9655-35c2-4465-8eb5-62843f37bb73)

2. As per the conversation we had on discord, I think everyone liked the idea of having the linked text in the section popup to say something like "<< View Course Page >>" Instead of having the code twice

Otherwise, i looked over everything and it looks great. Didnt observe any bugs or anything

ThatNerdSquared commented:
Updated the section popup:

![Screenshot 2024-06-17 at 13 51 11](https://github.com/mlool/workday-calendar-extension/assets/72814106/91e6d16e-145a-45b5-ab34-d284bb631d41)

Not sure if it makes sense to have it in such large font compared to the course name - I think this info view could use a little styling cleanup but that will have to wait for another PR.


The settings padding is, unfortunately, inconsistent across elements. I've added some to the Discord button for now but that whole pane could use a styling cleanup at some point.
![Screenshot 2024-06-17 at 13 51 24](https://github.com/mlool/workday-calendar-extension/assets/72814106/b50ed619-e78e-49bf-8b31-6f2f4c19d44b)


Also, I've included an updated `yarn.lock`, as yarn tells me to run `yarn install` when I build the extension for the first time. Let me know if this should be reverted.
